crates/pico/src/database.rs

```diff
@@ -16,6 +16,11 @@ use crate::derived_node::{DerivedNode, DerivedNodeId, DerivedNodeRevision};
 #[derive(Debug)]
 pub struct Database {
     pub(crate) dependency_stack: DependencyStack,
```

Comment:
Yeah, I like it. My initial thought about multithreading is that we should initialise the dependency stack from a top-level query or after Database cloning (assuming every top-level query is going to be in a separate thread). So even if `DependencyStack` may be something different in the future it is good to keep it separately from the storage.

Reply:
Yes, I'm very much on board with this